### PR TITLE
feat: Deprioritize ops methods in completion

### PR DESCRIPTION
**Pull request by Veykril**
Fixes https://github.com/rust-analyzer/rust-analyzer/issues/10593

### CODE REVIEW

**Comment by Veykril** [comment]
This currently slows down completions a lot because the lang_item queries aren't cheap and I think the cached values get lost pretty fast, at least the integrated completion benchmark is hitting it twice, even though the second time should be cached.

So I guess we want a better query for fetching the ops traits?

**Comment by Veykril** [comment]
Tried wrapping the lang item queries in an extra query which reduces the performance issues a bit, but still adds 7ms in the case where the query doesnt get reexecuted due to validation checking I suppose(hprof adds a `7ms ???` span). Any ideas on how to do this in a different way maybe?

**Comment by flodiebold** [comment]
I think you could just look for the `lang` attribute on the trait, instead of collecting all the lang items and checking whether it's one of them?

File: crates/ide_completion/src/context.rs [hunk 1]
```diff
@@ -308,6 +310,14 @@ impl<'a> CompletionContext<'a> {
         self.token.kind() == BANG && self.token.parent().map_or(false, |it| it.kind() == MACRO_CALL)
     }
 
+    /// Whether the given trait is an operator trait or not.
+    pub(crate) fn is_ops_trait(&self, trait_: hir::Trait) -> bool {
+        match trait_.attrs(self.db).lang() {
+            Some(lang) => OP_TRAIT_LANG_NAMES.contains(&lang.as_str()),
```
[the remark this answers is not in the document]
Comment (flodiebold) [reply crates/ide_completion/src/context.rs:316]: Yeah, I would think it's probably fine. Another option would be keeping the array sorted and doing a binary search in it.

**Comment by Veykril** [comment]
hmm this doesn't seem to work in vscode for me, maybe i missed something

**Pull request merged**